testing/trino-tests/src/test/java/io/trino/tests/TestLateMaterializationQueries.java

```diff
@@ -58,14 +63,20 @@ public void testTopN()
     @Test
     public void testSemiJoin()
     {
-        assertLazyQuery("SELECT * FROM orders WHERE orderkey IN (SELECT orderkey FROM orders WHERE orderdate < DATE '1994-01-01')");
```

Comment:
Does it make sense to keep the old test, or was it flaky?

if removal remains, please add explanation to the commit message

Reply:
It doesn't make sense to keep old test as the new one is superset of the previous one (two chained semi-joins)

Reply:
if this test case is still applicable, please keep it -- it's great to gave simple test cases before testing more complex ones